File: back/database/seeders/CoffeeSeeder.php
<?php

namespace Database\Seeders;

use App\Models\Coffee;
use Illuminate\Database\Seeder;

class CoffeeSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run(): void
    {
        Coffee::factory()->create([
            'type' => 'Espresso',
            'price' => 1.00,
            'image' => 'espresso.jpg',
            'coffee_amount' => 7,
            'brew_time' => 35
        ]);

        Coffee::factory()->create([
            'type' => 'Espresso doppio',
            'price' => 2.00,
            'image' => 'espresso_doppio.jpg',
            'coffee_amount' => 14,
            'brew_time' => 45
        ]);

        Coffee::factory()->create([
            'type' => 'Cappuccino',
            'price' => 2.50,
            'image' => 'cappuccino.jpg',
            'coffee_amount' => 7,
            'brew_time' => 60
        ]);
    }
}

File: src/main/java/com/example/coffeeshop/seeders/CoffeeSeeder.java
package com.example.coffeeshop.seeders;

import com.example.coffeeshop.models.Coffee;
import com.example.coffeeshop.repositories.CoffeeRepository;
import lombok.AllArgsConstructor;
import org.springframework.boot.CommandLineRunner;
import org.springframework.stereotype.Component;
import java.util.List;

@Component
@AllArgsConstructor
public class CoffeeSeeder implements CommandLineRunner {

    private final CoffeeRepository coffeeRepository;

    private void seedCoffees() {
        List<Coffee> coffeeList = List.of(
                new Coffee("Espresso", 35, 7, 1.00f, "espresso.jpg"),
                new Coffee("Espresso doppio", 45, 14, 2.00f, "espresso_doppio.jpg"),
                new Coffee("Cappuccino", 60, 7, 2.50f, "cappuccino.jpg")
        );
        this.coffeeRepository.saveAll(coffeeList);
    }

    @Override
    public void run(String... args) throws Exception {
        seedCoffees();
    }
}
